client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,461 @@
+                      newNonce <- Nonce.generate()
+                      _ <- transport.sendMessage(CreateSession(capabilities, newNonce)).orDie
+                    } yield newState.copy(nonce = newNonce)
+                  case None =>
+                    for {
+                      _ <- ZIO.logInfo("No more addresses, retrying from start")
+                      newNonce <- Nonce.generate()
+                      _ <- transport.sendMessage(CreateSession(capabilities, newNonce)).orDie
+                    } yield copy(currentAddressIndex = 0, nonce = newNonce)
+                }
+              
+              case SessionNotFound =>
+                ZIO.dieMessage("Session not found - cannot continue")
+              
+              case _ =>
+                ZIO.logWarning(s"Session rejected: $reason").as(Disconnected)
+            }
+          } else {
+            ZIO.logWarning("Nonce mismatch, ignoring SessionRejected").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(other) =>
+          ZIO.logWarning(s"Unexpected message in ConnectingNewSession: ${other.getClass.getSimpleName}").as(this)
+        
+        case StreamEvent.Action(ClientAction.SubmitCommand(_, promise)) =>
+          promise.fail(new IllegalStateException("Not connected")).ignore.as(this)
+        
+        case StreamEvent.Action(ClientAction.Disconnect) =>
+          ZIO.succeed(Disconnected)
+        
+        case _ =>
+          ZIO.succeed(this)
+      }
+    }
+    
+    def nextAddress: Option[(String, ConnectingNewSession)] = {
+      val nextIndex = currentAddressIndex + 1
+      if (nextIndex < addresses.length) {
+        Some((addresses(nextIndex), copy(currentAddressIndex = nextIndex)))
+      } else {
+        None
+      }
+    }
+  }
+  
+  /**
+   * Connecting to resume an EXISTING session (has sessionId).
+   */
+  case class ConnectingExistingSession(
+    sessionId: SessionId,
+    capabilities: Map[String, String],
+    nonce: Nonce,
+    addresses: List[String],
+    currentAddressIndex: Int,
+    nextRequestId: Ref[RequestId],
+    pendingRequests: Map[RequestId, Promise[Throwable, ByteVector]]
+  ) extends ClientState {
+    override def stateName: String = s"ConnectingExistingSession($sessionId)"
+    
+    override def handle(
+      event: StreamEvent,
+      transport: ZmqClientTransport,
+      config: ClientConfig,
+      serverRequestQueue: Queue[ServerRequest]
+    ): UIO[ClientState] = {
+      event match {
+        case StreamEvent.ServerMsg(SessionContinued(responseNonce)) =>
+          if (nonce == responseNonce) {
+            for {
+              _ <- ZIO.logInfo(s"Session continued: $sessionId")
+              now <- Clock.instant
+            } yield Connected(
+              sessionId = sessionId,
+              capabilities = capabilities,
+              createdAt = now,
+              serverRequestTracker = ServerRequestTracker(),
```

Comment:
we should not create a new serverRequestTracker when continuing a session, we must resume the previous one.